packages/next/src/server/stream-utils/node-web-streams-helper.test.ts

```diff
@@ -181,7 +182,37 @@ describe('node-web-stream-helpers', () => {
   describe('createBufferedTransformStream', () => {})
   describe('createInsertedHTMLStream', () => {})
   describe('renderToInitialFizzStream', () => {})
-  describe('createHeadInsertionTransformStream', () => {})
+  describe('createHeadInsertionTransformStream', () => {
+    it('should insert text into head tag', async () => {
+      const encoder = new TextEncoder()
+      const insertionInput = new ReadableStream({
+        start(controller) {
+          controller.enqueue('<foo>bar</foo>')
+          controller.close()
+        },
+      })
+      const reader = insertionInput.getReader()
+      async function insert() {
+        const { value } = await reader.read()
+        return value
+      }
+      const input = new ReadableStream({
+        start(controller) {
+          controller.enqueue(
+            encoder.encode('<head><fuzz>buzz</fuzz></head><body></body>')
+          )
```

Comment:
I'd add more than one chunk. maybe one for the head and then a couple more before closing the body. We'd expect to see insert called more than once because it will be called again on a future chunk after the current Task is complete. To model this you'll need to refactor to use `pull` so it starts only after the consumer is reading and you'll need to do the `await new Promise(r => setTimeout(r, 0))` trick